pkg/apis/k0s.k0sproject.io/v1beta1/clusterconfig_types.go

```diff
@@ -282,6 +282,16 @@ func (c *ClusterConfig) Validate() []error {
 	return errors
 }
 
+// CRValidator CRValidator is used to make sure a config CR is created with correct values
+func (c *ClusterConfig) CRValidator() *ClusterConfig {
+	copy := c.DeepCopy()
+	copy.ClusterName = "k0s"
+	copy.ObjectMeta.ClusterName = "k0s"
```

Comment:
This should be `copy.ObjectMeta.Name = "k0s"`

The `ClusterName` field in all kube objects is some weird historical left over AFAIK